src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/PkcsPalWindows.Encrypt.cs

```diff
@@ -207,36 +207,21 @@ public static SafeCryptMsgHandle CreateCryptMsgHandleToEncode(CmsRecipientCollec
             //
             private static CMSG_RECIPIENT_ENCODE_INFO EncodeRecipientInfo(CmsRecipient recipient, AlgorithmIdentifier contentEncryptionAlgorithm, HeapBlockRetainer hb)
             {
-                CMsgCmsRecipientChoice recipientChoice;
-                string keyAlgorithmOid = recipient.Certificate.GetKeyAlgorithm();
-                AlgId algId = keyAlgorithmOid.ToAlgId();
-                switch (algId)
-                {
-                    case AlgId.CALG_RSA_KEYX:
-                        recipientChoice = CMsgCmsRecipientChoice.CMSG_KEY_TRANS_RECIPIENT;
-                        break;
-
-                    case AlgId.CALG_DH_SF:
-                    case AlgId.CALG_DH_EPHEM:
-                        recipientChoice = CMsgCmsRecipientChoice.CMSG_KEY_AGREE_RECIPIENT;
-                        break;
-
-                    default:
-                        throw ErrorCode.CRYPT_E_UNKNOWN_ALGO.ToCryptographicException();
-                }
-
                 CMSG_RECIPIENT_ENCODE_INFO recipientEncodeInfo;
-                recipientEncodeInfo.dwRecipientChoice = recipientChoice;
 
                 unsafe
                 {
-                    switch (recipientChoice)
+                    switch (recipient.Certificate.GetKeyAlgorithm())
```

Comment:
`CertOIDToAlgId` will return `CALG_OID_INFO_PARAMETERS` here for RSAES_OAEP certificates, instead switch on the OID directly.